components/sendgrid/sendgrid.app.js

```diff
@@ -12,9 +13,17 @@ module.exports = {
     _apiUrl() {
       return "https://api.sendgrid.com/v3";
     },
-    _contactsSearchUrl() {
+    api() {
+      sendgrid.setDefaultHeader({
+        "Content-Type": "application/json",
+        "User-Agent": "@PipedreamHQ/pipedream v0.1",
+      });
+      sendgrid.setApiKey(this.$auth.api_key);
+      return sendgrid;
+    },
```

Comment:
Might as well use `_authToken()` for getting `this.$auth.api_key`.
```suggestion
    api() {
      sendgrid.setDefaultHeader({
        "Content-Type": "application/json",
        "User-Agent": "@PipedreamHQ/pipedream v0.1",
      });
      sendgrid.setApiKey(this._authToken());
      return sendgrid;
    },
```